packages/ui-app/src/InputNumber.tsx

```diff
@@ -0,0 +1,159 @@
+// Copyright 2017-2018 @polkadot/ui-app authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { TranslationFunction } from 'i18next';
+import { RawParam$OnChange } from './Params/types';
+import { BareProps } from './types';
+
+import BN from 'bn.js';
+import React from 'react';
+
+import isValidBalance from './util/isValidBalance';
+import { keydown } from './util/keyboard';
+import classes from './util/classes';
+import Input from './Input';
+import { KEYS_ALLOWED, KEYS_PRE } from './constants';
+
+type Props = BareProps & {
+  defaultValue?: string,
+  isError?: boolean,
+  label?: any,
+  maxLength?: number,
+  onChange?: RawParam$OnChange | undefined,
```

Comment:
This should just be a `(value: BN) => void` callback. The Param stuff is only for Params. (e.g. see other Inputs, it only returns the value)

Reply:
I've had to change it to `onChange?: (value: { isValid: boolean, value: BN }) => void,` instead of `onChange?: (value: RawParam$OnChange$Value) => void,` (that way we're not using Param types), and because the onChange that is passed as a prop to InputNumber from the Balance component comes from BareProps and is of type `RawParam$OnChange | undefined`, and the `onChange` method that we call on InputNumber returns `{ isValid, value: valueBN }`